core/ts/MaciState.ts

```diff
@@ -430,17 +440,21 @@ class Poll {
             let pubKey: any;
 
             if (computedStateIndex > -1) {
-                pubKey = this.stateLeaves[computedStateIndex].pubKey;
+                pubKey = this.stateLeaves[stateIndexInt].pubKey;
+                stateLeafPathElements.push(this.stateTree.genMerklePath(stateIndexInt).pathElements);
+                currentStateLeaves.push(this.stateLeaves[stateIndexInt].asCircuitInputs());
             } else {
                 pubKey = new PubKey([BigInt(0), BigInt(0)]);
+                stateLeafPathElements.push(this.stateTree.genMerklePath(0).pathElements);
+                currentStateLeaves.push(this.stateLeaves[0].asCircuitInputs());
             }
 
             // Verify deactivation message
             const status = deactCommand.cmdType.toString() == '1' // Check message type
                 && computedStateIndex != -1
                 && signature != null
                 && verifySignature(
-                    deactMessage.hash(encPubKey),
+                    deactCommand.hash(),
```

Comment:
`verifySignature` seems to take a message as first parameter. The command hash should be different from the signature. Could you please check that you are passing the correct value?

Reply:
msg param from VerifySignature is just a BigInt, so a hash is valid. Old param was also just a hash. This change was introduced by Aleksandar and i believe was changed because the previous one was either unnecessary or plain wrong. Tests pass so we can verify with Aleksandar once he is back,

Reply:
Let me rephrase the question: I was wondering since `verifySignature` takes `message` as its first parameter - how come we're passing a `command` hash. Is this to be consider correct or not? If you'd like, we could punt back to when Alex is back